### DISCUSSION TITLE
How is it forecasting if future features are unknown, Isnt it just regression?

### DISCUSSION THREAD

**Discussion by f-izzat** [blue-yonder/tsfresh#964] [Q&A]
- Today is 2022-08-30, I have the `high` value for this.
- I have followed the Forecasting stocks notebook
- Say I would like to forecast the value for 2022-08-31, **but I dont have the value for this in order to extract my X features**
- Why would I want a model to predict the `high` value for today when I already have it?
- Title for forecasting should be changed to 'regression' unless there is a way to **predict future values and not just today**

**Comment by nils-braun** [comment]
I assume you are referring to [this](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/05%20Timeseries%20Forecasting.ipynb) notebook. The procedure described there is used to predict the value on `2022-08-31` using data up to and including `2022-08-30`. The notebook is using  `shift` operation to accomplish this.
Of course, it is still a regression (so you are right), because the target is real-valued (and not a classification problem). It is however also a prediction because I am using values from the past to predict values of the future. You could change the shift parameter from -1 to -7 to try predicting the values 7 days into the future.
